### PR TITLE
Icon: pass props to <i>

### LINKED ISSUE #170: Icon组件如何自定义样式?
除了HOC方法外, 希望能提供这样的快捷方式: `<Icon type="edit" style={{color: "green"}} />`

ant-design里面已经有了https://github.com/ant-design/ant-design/issues/2863

### PR DESCRIPTION
**Pull request by gutenye**
fixes #170

### CODE REVIEW

File: components/icon/index.web.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default class Icon extends React.Component<IconPropType, any> {
   render() {
     const { type, className } = this.props;
     return (
-      <i className={`${className} anticon anticon-${type}`.trim()} />
+      <i {...this.props} className={`${className} anticon anticon-${type}`.trim()} />
```
Comment (benjycui) [inline components/icon/index.web.tsx]: https://facebook.github.io/react/warnings/unknown-prop.html

Comment (gutenye) [inline components/icon/index.web.tsx]: thanks, fixed.

File: components/icon/index.web.tsx [hunk 2]
```diff
@@ -10,9 +10,9 @@ export default class Icon extends React.Component<IconPropType, any> {
     className: '',
   };
   render() {
-    const { type, className } = this.props;
+    const { type, className, ...rest } = this.props;
```
Comment (silentcloud) [inline components/icon/index.web.tsx]: ts 没过, npm run compile

**Comment by warmhug** [comment]
已修复了，谢谢

**Pull request closed without merging**